Enterprise-CMCS/managed-care-review
# MR-3428 insertContract returns unlocked contract

## Summary
[MR-3428](https://qmacbis.atlassian.net/jira/software/projects/MR/boards/234?selectedIssue=MR-3428)

Follow up to the database work.

#### Related issues

#### Screenshots

#### Test cases covered

- `insertContract.test.ts`
   - `'creates a new draft contract'`



## QA guidance

- Inserting a new draft contract was simple. The parts that need the most scrutiny are:
   - Refactors in Prisma queries
   - `contractFormDataToDomainModel` function to make sure we have correct domain data
   - `prismaTypes.ts` has a new `ContractRevisionTableWithRelations` type used for contract revisions with all relationships. I couldn't a type like this from `@prisma/client`
   - `prismaTypes.ts` has `draftContractRevisionsWithDraftRates` used for Prisma query on draft contracts.
